Frontend/ui-customer/src/Screens/CustomerUtils/ReportIssueChatScreen/Controller.purs

```diff
@@ -0,0 +1,475 @@
+    _ -> trackAppScreenEvent appId (getScreen REPORT_ISSUE_CHAT_SCREEN) "in_screen" "other_action"
+
+instance showAction :: Show Action where
+    show _ = ""
+
+data Action = Exit (ScreenOutput)
+            | NoAction
+            | BackPressed
+            | AfterRender
+            | SubmitIssue
+            | ShowOptions
+            | AddImage
+            | AddAudio Boolean
+            | DeleteRecordedAudio
+            | DeleteSelectedImages
+            | CancelCall  PrimaryButton.Action
+            | ConfirmCall PrimaryButton.Action
+            | AddAudioModelAction  AddAudioModel.Action
+            | AddImagesModelAction AddImagesModel.Action
+            | ImageUploadCallback  String String String
+            | UploadMultiPartDataCallback  String String
+            | ViewImageModelAction ViewImageModel.Action
+            | RecordAudioModelAction   RecordAudioModel.Action
+            | UpdateRecordModelPlayer  String
+            | ChatViewActionController ChatView.Action
+            | SendMessage ChatView.ChatComponentConfig Boolean
+            | UpdateState ReportIssueChatScreenState
+            | PrimaryEditTextActionController PrimaryEditText.Action
+            | ReopenIssuePress
+            | KeyboardCallback String
+            | InitializeCallback
+
+data ScreenOutput = GoBack
+                  | UploadIssue  ReportIssueChatScreenState
+                  | CallDriver ReportIssueChatScreenState
+                  | CallSupport ReportIssueChatScreenState
+                  | SelectIssueOption ReportIssueChatScreenState
+                  | ReopenIssue ReportIssueChatScreenState
+
+eval :: Action -> ReportIssueChatScreenState -> Eval Action ScreenOutput ReportIssueChatScreenState
+
+eval ReopenIssuePress state = 
+  exit $ ReopenIssue state
+
+eval (Exit output) state =
+  exit output
+
+eval (KeyboardCallback keyBoardState) state = do 
+  let keyState = case keyBoardState of
+                    "onKeyboardOpen" -> true
+                    "onKeyboardClose" -> false
+                    _ -> false 
+  _ <- pure $ scrollToEnd (getNewIDWithTag "ChatScrollView") true
+  continue state {props {isKeyboardOpen = keyState}}
+
+eval BackPressed state =
+  continueWithCmd state [do
+    _ <- runEffectFn1 removeMediaPlayer ""
+    _ <- pure $ hideKeyboardOnNavigation true
+    pure $ Exit GoBack
+  ]
+
+eval AfterRender state =
+  continue state
+
+eval SubmitIssue state =
+  exit $ UploadIssue state
+
+eval ShowOptions state = do
+  let options'  = map (\x -> x.option) state.data.options
+  continue state { data { chatConfig { enableSuggestionClick = false, suggestionsList = options' } } }
+
+---------------------------------------------------- Add Media ----------------------------------------------------
+eval AddImage state =
+  if length state.data.addedImages > 0
+  then
+    continueWithCmd state { props { showImageModel = true, isPopupModelOpen = true }
```

Comment:
Looks like some state is similar in both cases
Create a new var with common state and use it